app/assets/javascripts/covid19/src/webapp/components/structures-table/IDRDialog.tsx

```diff
@@ -0,0 +1,210 @@
+import _ from "lodash";
+import React from "react";
+import styled from "styled-components";
+import { Typography } from "@material-ui/core";
+import { Pagination } from "@material-ui/lab";
+import { ListItemProps } from "./cells/DetailsCell";
+import { colors } from "./badge/Badge";
+import { Dialog } from "./Dialog";
+import { IDROptions } from "./Columns";
+import { Assay, Compound, Screen } from "../../../domain/entities/LigandImageData";
+import i18n from "../../../utils/i18n";
+
+export interface IDRDialogProps {
+    onClose(): void;
+    idrOptions?: IDROptions;
+    open: boolean;
+}
+
+export const IDRDialog: React.FC<IDRDialogProps> = React.memo(props => {
+    const { onClose, open } = props;
+    const { ligand, idr } = props.idrOptions ?? { ligand: undefined, idr: undefined };
+    const [page, setPage] = React.useState(1);
+    const handleChange = React.useCallback((_event: React.ChangeEvent<unknown>, value: number) => {
+        setPage(value);
+    }, []);
+
+    return (
+        <StyledDialog
+            open={open}
+            onClose={onClose}
+            title={ligand?.name ? `${ligand.name} (${ligand.id})` : i18n.t("Ligand IDR")}
+            maxWidth={"sm"}
+        >
+            <Wrapper>
+                <Container>
+                    {idr?.assays.map((assay, idx) => {
```

Comment:
Not sure, so it seems an empty array for idr removes the need of `?` ?

Reply:
I don't understand very well what you are saying. But assays is always an array, don't know why ts marks as possibly undefined when idr is not undefined